core/drivers/scmi-msg/entry.c

```diff
@@ -13,6 +18,65 @@
 #include "reset_domain.h"
 #include "voltage_domain.h"
 
+#define SCMI_PLAYLOAD_U32_MAX		(SCMI_PLAYLOAD_MAX / sizeof(uint32_t))
+
+/* Provision input message payload buffers for each supported entry */
+static uint32_t fast_smc_payload[CFG_TEE_CORE_NB_CORE][SCMI_PLAYLOAD_U32_MAX]
+__maybe_unused;
+
+static uint32_t interrupt_payload[CFG_TEE_CORE_NB_CORE][SCMI_PLAYLOAD_U32_MAX]
+__maybe_unused;
+
+static uint32_t threaded_payload[CFG_NUM_THREADS][SCMI_PLAYLOAD_U32_MAX]
+__maybe_unused;
+
+/* SMP protection on channel access */
+static unsigned int smt_channels_lock;
+
+/* If channel is not busy, set busy and return true, otherwise return false */
+bool scmi_channel_set_busy(struct scmi_msg_channel *channel)
+{
+	uint32_t exceptions = cpu_spin_lock_xsave(&smt_channels_lock);
+	bool channel_is_busy = channel->busy;
+
+	if (!channel_is_busy)
+		channel->busy = true;
+
+	cpu_spin_unlock_xrestore(&smt_channels_lock, exceptions);
+
+	return !channel_is_busy;
+}
+
+void scmi_channel_release_busy(struct scmi_msg_channel *channel)
+{
+	channel->busy = false;
+}
+
+#ifdef CFG_SCMI_MSG_SMT_FASTCALL_ENTRY
```

Comment:
Do we need all these ifdefs, can't we rely on the linker to garbage collect unused functions?

Reply:
ok